moto/config/models.py

```diff
@@ -411,13 +452,403 @@ def update(
+            if message_type in [
+                "ConfigurationItemChangeNotification",
+                "OversizedConfigurationItemChangeNotification",
+            ]:
+                raise InvalidParameterValueException(
+                    "A maximum execution frequency is not allowed if "
+                    "MessageType is ConfigurationItemChangeNotification or "
+                    "OversizedConfigurationItemChangeNotification"
+                )
+        else:
+            # If no value is specified, use a default value for
+            # maximum_execution_frequency for message types representing a
+            # periodic trigger.
+            if message_type in [
+                "ScheduledNotification",
+                "ConfigurationSnapshotDeliveryCompleted",
+            ]:
+                maximum_execution_frequency = SourceDetail.DEFAULT_FREQUENCY
+
+        self.event_source = event_source
+        self.message_type = message_type
+        self.maximum_execution_frequency = maximum_execution_frequency
+
+
+class Source(ConfigEmptyDictable):
+
+    """Defines rule owner, id and notification for triggering evaluation."""
+
+    OWNERS = {"AWS", "CUSTOM_LAMBDA"}
+
+    def __init__(self, region, owner, source_identifier, source_details=None):
+        super().__init__(capitalize_start=True, capitalize_arn=False)
+        if owner not in Source.OWNERS:
+            raise ValidationException(
+                f"Value '{owner}' at 'configRule.source.owner' failed to "
+                f"satisfy constraint: Member must satisfy enum value set: {{"
+                + ", ".join(sorted(Source.OWNERS))
+                + "}"
+            )
+
+        if owner == "AWS":
+            # Can the Source ID be found in the dict of managed rule IDs?
+            if source_identifier not in MANAGED_RULES_CONSTRAINTS:
+                raise InvalidParameterValueException(
+                    f"The sourceIdentifier {source_identifier} is invalid.  "
+                    f"Please refer to the documentation for a list of valid "
+                    f"sourceIdentifiers that can be used when AWS is the Owner"
+                )
+            if source_details:
+                raise InvalidParameterValueException(
+                    "SourceDetails should be null/empty if the owner is AWS. "
+                    "SourceDetails should be provided if the owner is "
+                    "CUSTOM_LAMBDA"
+                )
+
+            self.owner = owner
+            self.source_identifier = source_identifier
+            self.source_details = None
+            return
+
+        # Otherwise, owner == "CUSTOM_LAMBDA"
+        if not source_details:
+            raise InvalidParameterValueException(
+                "SourceDetails should be null/empty if the owner is AWS. "
+                "SourceDetails should be provided if the owner is CUSTOM_LAMBDA"
+            )
+
+        lambda_func = lambda_backends[region].get_function(source_identifier)
```

Comment:
Can we move the `lambda_backends` import down to this level?
It's quite a slow operation, and it imports some dependencies that might not be necessary for all Config-users, so we should only load it if/when required